distributed/tests/test_scheduler.py

```diff
@@ -716,6 +716,7 @@ def test_retire_workers(c, s, a, b):
     workers = yield s.retire_workers()
     assert not workers
 
+    assert all(v==0 for v in s.suspicious_tasks.values())
```

Comment:
This probably would have been this way regardless.  I think that we need a task to be actively running when we take down the worker.  In this case we just have data sitting there but no tasks running.

Maybe something something like the following?

```python
futures = c.map(slowinc, range(10), delay=0.05, workers=b.address)
future = c.submit(slowinc, 100, delay=0.5, workers=a.address, allow_other_workers=True)
yield gen.sleep(0.2)
yield s.retire_workers()  # I'm hoping that this closes worker a

assert all(v==0 for v in s.suspicious_tasks.values())
```

Although I suppose that if you're also going to stop `retire_workers` from closing workers that have running tasks then this might go away anyway?

Reply:
Yes and no. That change will make this go away in the sense that `retire_workers()` should no longer choose to retire worker a, but we can force this in this test with `yield s.retire_workers(workers=a.address)`. This is leaving open the possibility that `retire_workers` gets called by some other function either in the future or in some sub-class.

Reply:
OK great, I had forgotten about the ability to force retirement.  Presumably the example above could be simplified to something like the following:

```python
future = c.submit(slowinc, 100, delay=0.5, workers=a.address, allow_other_workers=True)
yield gen.sleep(0.2)
yield s.retire_workers(workers=[a.address])

assert all(v==0 for v in s.suspicious_tasks.values())
```